osu.Game.Rulesets.Taiko/UI/HitExplosion.cs

```diff
@@ -37,35 +34,36 @@ public HitExplosion(DrawableHitObject judgedObject, HitType type)
             Size = new Vector2(TaikoHitObject.DEFAULT_SIZE);
 
             RelativePositionAxes = Axes.Both;
-
-            BorderColour = Color4.White;
-            BorderThickness = 1;
-
-            Alpha = 0.15f;
-            Masking = true;
-
-            Children = new[]
-            {
-                innerFill = new Box
-                {
-                    RelativeSizeAxes = Axes.Both,
-                }
-            };
         }
 
         [BackgroundDependencyLoader]
-        private void load(OsuColour colours)
+        private void load()
         {
-            innerFill.Colour = type == HitType.Rim ? colours.BlueDarker : colours.PinkDarker;
+            Child = new SkinnableDrawable(new TaikoSkinComponent(getComponentName(JudgedObject.Result?.Type ?? HitResult.Great)), _ => new DefaultHitExplosion());
+        }
+
+        private TaikoSkinComponents getComponentName(HitResult resultType)
+        {
+            switch (resultType)
+            {
+                case HitResult.Miss:
+                    return TaikoSkinComponents.TaikoExplosionMiss;
+
+                case HitResult.Good:
+                    return TaikoSkinComponents.TaikoExplosionGood;
+
+                case HitResult.Great:
+                    return TaikoSkinComponents.TaikoExplosionGreat;
+            }
+
+            throw new ArgumentException("Invalid result type", nameof(resultType));
```

Comment:
```suggestion
            throw new ArgumentOutOfRangeException(nameof(resultType), "Invalid result type");
```
fits a little better IMO.